tests/contrib/io/matplotlib/test_matplotlib_s3_writer.py

```diff
@@ -0,0 +1,238 @@
+                "Condition": {"Null": {"s3:x-amz-server-side-encryption": "aws:kms"}},
+            }
+        ],
+    }
+    bucket_policy = json.dumps(bucket_policy)
+
+    with mock_s3():
+        conn = s3fs.core.boto3.client("s3", **AWS_CREDENTIALS)
+        conn.create_bucket(Bucket=BUCKET_NAME)
+        conn.put_bucket_policy(Bucket=BUCKET_NAME, Policy=bucket_policy)
+
+        yield conn
+
+
+@pytest.fixture
+def plot_writer(mocked_s3_bucket):  # pylint: disable=unused-argument
+    def _matplotlibwriters3():
+        return MatplotlibS3Writer(
+            bucket_name=BUCKET_NAME, filepath=KEY_PATH, credentials=AWS_CREDENTIALS
+        )
+
+    return _matplotlibwriters3()
+
+
+def test_save_data(tmp_path, mock_single_plot, plot_writer, mocked_s3_bucket):
+    """Test saving single matplotlib plot to S3."""
+
+    plot_writer.save(mock_single_plot)
+
+    expected_path = tmp_path / "downloaded_image.png"
+    actual_filepath = tmp_path / "locally_saved.png"
+
+    plt.savefig(str(actual_filepath))
+
+    mocked_s3_bucket.download_file(BUCKET_NAME, KEY_PATH, str(expected_path))
+
+    assert actual_filepath.read_bytes() == expected_path.read_bytes()
+
+
+def test_list_save(tmp_path, mock_list_plot, plot_writer, mocked_s3_bucket):
+    """Test saving list of plots to S3."""
+
+    plot_writer.save(mock_list_plot)
+
+    for index in range(5):
+
+        expected_path = tmp_path / "downloaded_image.png"
+        actual_filepath = tmp_path / "locally_saved.png"
+
+        mock_list_plot[index].savefig(str(actual_filepath))
+
+        _key_path = KEY_PATH + "/" + str(index) + ".png"
+
+        mocked_s3_bucket.download_file(BUCKET_NAME, _key_path, str(expected_path))
+
+        assert actual_filepath.read_bytes() == expected_path.read_bytes()
+
+
+def test_dict_save(tmp_path, mock_dict_plot, plot_writer, mocked_s3_bucket):
+    """Test saving dictionary of plots to S3."""
+
+    plot_writer.save(mock_dict_plot)
+
+    for colour in COLOUR_LIST:
+
+        expected_path = tmp_path / "downloaded_image.png"
+        actual_filepath = tmp_path / "locally_saved.png"
+
+        mock_dict_plot[colour].savefig(str(actual_filepath))
+
+        _key_path = KEY_PATH + "/" + colour
+
+        mocked_s3_bucket.download_file(BUCKET_NAME, _key_path, str(expected_path))
+
+        assert actual_filepath.read_bytes() == expected_path.read_bytes()
+
+
+def test_bad_credentials(mock_dict_plot):
+    """Test writing with bad credentials"""
+    bad_writer = MatplotlibS3Writer(
+        bucket_name=BUCKET_NAME,
+        filepath=KEY_PATH,
+        credentials={
+            "aws_access_key_id": "not_for_testing",
+            "aws_secret_access_key": "definitely_not_for_testing",
+        },
+    )
+    pattern = "InvalidAccessKeyId"
+
+    with pytest.raises(DataSetError, match=pattern):
+        bad_writer.save(mock_dict_plot)
+
+
+def test_credentials(tmp_path, mock_single_plot, mocked_s3_bucket):
```

Comment:
Looks like this test is doing exactly the same as `test_save_data` above.

Reply:
I would also test `test_s3fs_args` as an additional test

Reply:
`test_s3_encryption` below actually covers using `s3fs_args`. Should I just rename from `test_credentials` to `test_s3fs_args`?

Reply:
Renaming from `test_s3_encryption` to `test_s3fs_args`.

Tbh, I cannot think of any other arg to test, nor whether `moto` only mocks the calls and not the actual s3 properties.